File: gluecodium/src/test/resources/smoke/attributes/output/android/com/example/smoke/AttributesStruct.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
@OnStruct
public final class AttributesStruct {
    @OnConstInStruct
    public static final boolean PI = false;
    @OnField
    @NonNull
    public String field;
    public AttributesStruct(@NonNull final String field) {
        this.field = field;
    }
    @OnFunctionInStruct
    public native void veryFun(@OnParameterInStruct @NonNull final String param);
}

File: gluecodium/src/test/resources/smoke/attributes/output/android-kotlin/com/example/smoke/AttributesStruct.kt
/*

 *
 */

@file:JvmName("AttributesStruct")

package com.example.smoke


class AttributesStruct {
    @JvmField var field: String



    constructor(field: String) {
        this.field = field
    }





    external fun veryFun(param: String) : Unit


    companion object {
        @OnConstInStruct
        @JvmField final val PI: Boolean = false
    }
}

